h2o-r/h2o-package/R/frame.R

```diff
@@ -3291,14 +3296,13 @@ as.h2o.data.frame <- function(x, destination_frame="", ...) {
 #' Turn on h2o datatable by options("h2o.use.data.table"=TRUE)
 #' @export
 as.h2o.Matrix <- function(x, destination_frame="", ...) {
-  
-  if( destination_frame=="")
-    destination_frame <- deparse(substitute(x))
+  if( destination_frame=="") {
+    subx <- destination_frame.guess(deparse(substitute(x)))
+    destination_frame <- .key.make(if(nzchar(subx)) subx else "Matrix")
+  }
 
   destination_frame <- destination_frame.guess(destination_frame) # filter out invalid i.e. "abc::fun()"
```

Comment:
this line could be in `else {` block so we won't repeat same call unnecessarily.

Reply:
adding it